Track analytic event when editor help is shown

dcalhoun opened the pull request:
## Related PRs

* https://github.com/wordpress-mobile/WordPress-iOS/pull/17045

## Description

Improve our ability to understand how users interact with the editor help content.

To test:

1. Launch the editor.
1. Tap the three-dot button in the top right corner of the editor.
1. Tap Help.
1. ℹ️ **Expected:** The `wpandroid_editor_help_shown` event arrives in Tracks.

## Regression Notes
1. Potential unintended areas of impact
  The new analytic event could disrupt opening the help content.
2. What I did to test those areas of impact (or what existing automated tests I relied on)
  Verified the help content continues to display.
3. What automated tests I added (or what prevented me from doing so)
  The majority of the help content logic resides in the editor, so placing tests
within this repository makes less sense.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

dcalhoun committed:
Track analytic event when editor help is shown

Improve our ability to understand how users interact with the editor
help content.

mchowning reviewed (approved):
Changes look good and I'm seeing the events show up in Tracks. Nice job!

Don't forget to validate the new event in Tracks. Could also register it, but that is substantially more work and my understanding is that people frequently bypass registering the event.

dcalhoun committed:
Set blogid property for editor_help_shown event

Documentation for validating event states all events that occur in a
site context should have this property set.

mchowning committed:
Use trackWithSiteDetails method

mchowning reviewed (changes requested):
Thanks for making this update @dcalhoun ! I see the blog id coming through in analytics now. 😄 

I do think we can improve this a bit. I don't think we need to use the `onTrackableEvent` method here because it looks like that is really used for the purpose of processing trackable events that are originating within the editor module because that module does not have access to the main app module's `AnalyticsUtils`. Here, though, we can directly access the `AnalyticsUtils` using the wrapper (`mAnalyticsTrackerWrapper`).

It also really stuck out to me that we shouldn't have to manually add the blog id. I dug a bit and the `AnalyticsUtils` has a method that handles that for us: [AnalyticsUtils::trackWithSiteDetails](https://github.com/wordpress-mobile/WordPress-Android/blob/db0caa32372ff5ffde69beb32ad7abdaec6717a5/WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java#L256). Unfortunately, that method isn't exposed in the `AnalyticsTrackerWrapper` the way we need it, so we'll need to add that method. 

I pushed [a commit](https://github.com/wordpress-mobile/WordPress-Android/commit/a6eb07a2947844033c3cf55017e6ad783943a027) with what I'm thinking may be a good way to go about this. I did make a few extra changes in that commit to improve the null-safety of the code in this area. If you'd rather just stick to the changes that are strictly necessary for this PR, that's fine. Let me know what you think.

dcalhoun committed:
Remove redundant function

The AnalyticsTrackerWrapper already includes same method signature under
the name `track`.

dcalhoun commented:
> I don't think we need to use the `onTrackableEvent` method here because [...] we can directly access the `AnalyticsUtils` using the wrapper (`mAnalyticsTrackerWrapper`).

@mchowning aha! Yes, that makes sense. I had too narrow a focus when refactoring, which caused me to overlook this aspect. Thank you for calling this out. 

> I pushed [a commit](https://github.com/wordpress-mobile/WordPress-Android/commit/a6eb07a2947844033c3cf55017e6ad783943a027) with what I'm thinking may be a good way to go about this.

Thanks for this. FYI the commit you linked appears to be detached from a branch and missing the last change you made to `MySiteFragment.kt` resolving a build error. I believe I found the correct https://github.com/wordpress-mobile/WordPress-Android/commit/7a2bd246a3db28343a69493cd2a2ff34257af17e commit, and have cherry picked that into this branch. 

> I did make a few extra changes in that commit to improve the null-safety of the code in this area. If you'd rather just stick to the changes that are strictly necessary for this PR, that's fine. Let me know what you think.

All of the changes make to sense to me thanks to your explanation. I reviewed and tested the code again; I think this is an improved approach. I did notice that `AnalyticsTrackerWrapper.kt` already possessed a method with a similar signature. 

https://github.com/wordpress-mobile/WordPress-Android/blob/fba084cd05be8b751f17085db4f7a577b7edee58/WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsTrackerWrapper.kt#L45-L47

So, I pushed https://github.com/wordpress-mobile/WordPress-Android/pull/15219/commits/45e5d8babc69234c63a200926443eedba2cb23bf leveraging the existing `track` method instead of adding `trackWithSiteDetails`. Do you think that makes sense or should we retain the new `trackWithSiteDetails` method added in your previous commit?

mchowning commented:
> I believe I found the correct 7a2bd24 commit, and have cherry picked that into this branch.

You're absolutely right, I linked the wrong commit. Sorry about that.

> So, I pushed 45e5d8b leveraging the existing track method instead of adding trackWithSiteDetails. Do you think that makes sense or should we retain the new trackWithSiteDetails method added in your previous commit?

👍  Good catch, I totally missed that.

mchowning reviewed (approved):
Looks good and is working well! :shipit:

dcalhoun committed:
Fix line length lint error
